lte/cloud/go/services/subscriberdb/storage/subscriber_state.go

```diff
@@ -182,6 +195,58 @@ func (ss *subscriberStorage) DeleteSubscribersForGateway(networkID string, gatew
 	return err
 }
 
+// GetSubscribersForIMSIs takes a network ID and a list of IMSIs and returns the
+// subscriber states of those IMSIs in the network.
+// If imsis == nil, states for all IMSIs are returned.
+func (ss *subscriberStorage) GetSubscribersForIMSIs(networkID string, imsis []string) (ImsiStateMap, error) {
+	txFn := func(tx *sql.Tx) (interface{}, error) {
+		var rows *sql.Rows
+		var err error
+		if imsis == nil {
+			rows, err = ss.getSubscribersForNetwork(tx, networkID)
+		} else {
+			rows, err = ss.getSubscribersForIMSIs(tx, networkID, imsis)
+		}
+		if err != nil {
+			return nil, fmt.Errorf("Error getting subscribers for nid / imsis:  %v / %v", networkID, imsis)
+		}
+		defer sqorc.CloseRowsLogOnError(rows, "GetSubscribersForIMSIs")
+
+		mappings := ImsiStateMap{}
+		lastUpdates := map[string]int64{}
+		for rows.Next() {
+			var rawState []byte
+			imsi := ""
+			var lastUpdated int64
+			err = rows.Scan(&imsi, &rawState, &lastUpdated)
+			if err != nil {
+				return nil, fmt.Errorf("GetSubscribersForIMSIs, SQL row scan error: %w", err)
+			}
+			reportedState := make(state.ArbitraryJSON)
+			err = json.Unmarshal(rawState, &reportedState)
+			if err != nil {
+				return nil, fmt.Errorf("GetSubscribersForIMSIs, error unmarshaling state: %w", err)
+			}
+			if lastUpdated > lastUpdates[imsi] {
+				lastUpdates[imsi] = lastUpdated
+				mappings[imsi] = reportedState
+			}
+		}
+		err = rows.Err()
+		if err != nil {
+			return nil, fmt.Errorf("GetSubscribersForIMSIs, SQL rows error: %w", err)
+		}
```

Comment:
Idea: Maybe this could be extracted to a new method `convertRowsToMap` (or similar).